### PR TITLE
db: add missing Close()'s

### PR DESCRIPTION
**Pull request by dajohi**

### CODE REVIEW

File: db/dcrpg/queries.go [hunk 1]
```diff
@@ -783,6 +785,7 @@ func retrieveTimeBasedBlockListing(ctx context.Context, db *sql.DB, timeInterval
 	if err != nil {
 		return nil, fmt.Errorf("retrieveTimeBasedBlockListing failed: error: %v", err)
 	}
+	defer closeRows(rows)
```
Comment (chappjc) [inline db/dcrpg/queries.go]: Wow.  I can't believe we missed this one

File: db/dcrpg/queries.go [hunk 2]
```diff
@@ -1398,7 +1399,7 @@ func RetrieveAddressTxnsOrdered(ctx context.Context, db *sql.DB, addresses []str
 
 	var rows *sql.Rows
 	rows, err = stmt.QueryContext(ctx, pq.Array(addresses))
-	// _ = stmt.Close() // or does Rows.Close do it?
```
[the remark this answers is not in the document]
Comment (dajohi) [reply db/dcrpg/queries.go]: docs say `The caller must call the statement's Close method when the statement is no longer needed. `  

I can't find anything about rows.Close() doing it.

http://go-database-sql.org/retrieving.html examples close stmt and rows explicitly.

File: db/dcrpg/queries.go [hunk 3]
```diff
@@ -2299,6 +2292,7 @@ func SetSpendingForVinDbIDs(db *sql.DB, vinDbIDs []uint64) ([]int64, int64, erro
 		err = vinGetStmt.QueryRow(vinDbID).Scan(
 			&txHash, &txVinInd, &prevOutHash, &prevOutVoutInd)
 		if err != nil {
+			bail()
 			return addressRowsUpdated, 0, fmt.Errorf(`SelectVinVoutPairByID: `+
 				`%v + %v (rollback)`, err, bail())
```
Comment (dajohi) [inline db/dcrpg/queries.go]: oh, bail() is in the fmt.Errorf.  let me fix.

File: db/dcrpg/queries.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (chappjc) [reply db/dcrpg/queries.go]: Cool.  We must have been leaking some resources.

**Pull request merged**